packages/hardhat-etherscan/src/index.ts

```diff
@@ -85,6 +85,7 @@ interface VerificationSubtaskArgs {
   // Fully qualified name of the contract
   contract?: string;
   libraries: Libraries;
+  noCompile?: boolean;
```

Comment:
I think the type of this property is just `boolean`, not `boolean | undefined`, because flags default to false if not provided. This should also simplify the logic where this value is used.

You can check the [test task](https://github.com/NomicFoundation/hardhat/blob/master/packages/hardhat-core/src/builtin-tasks/test.ts) to see how it's done there.